main/networking/Runtime.ts

```diff
@@ -77,27 +78,160 @@ function createPacket(payload: unknown, messageType: MsgType): Buffer {
     case MsgType.CHALLENGE_DATA:
       encodedPayload = protos.Text.encode(payload as protos.IText).finish();
       break;
+    case MsgType.INPUTS:
+      encodedPayload = protos.Text.encode(payload as any).finish();
+      break;
     default:
       console.log('ERROR: trying to create TCP Packet with type LOG');
       encodedPayload = new Uint8Array();
       break;
   }
-  
+
   const msgLength = Buffer.byteLength(encodedPayload);
   const msgLengthArr = new Uint8Array([msgLength & 0x00ff, msgLength & 0xff00]); // Assuming little-endian byte order, since runs on x64
   const msgTypeArr = new Uint8Array([messageType]);
 
   return Buffer.concat([msgTypeArr, msgLengthArr, encodedPayload], msgLength + 3);
 }
 
+class BaseTCPConn {
+  connectionName: string;
+  logger: Logger;
+  tcpSocket: TCPSocket;
+  ip: string;
+  port: number;
+  onConnect?: () => void;
+  onConnectionClose?: () => void;
+  onDataReceived: (data: Buffer) => void;
+
+  constructor({
+    connectionName,
+    logger,
+    onDataReceived,
+    onConnect,
+    onConnectionClose,
+    ip = defaults.IPADDRESS,
+    port = DEFAULT_TCP_PORT
+  }: {
```

Comment:
is there a reason you used this object as the parameter rather than just having all the params in the constructor?

Reply:
No reason in particular, just thought it would be easier to add and remove parameters from the object

Reply:
I think it looks nicer if you just kept them as normal parameters. You can have default undefined parameters for `onConnectionClose` and `onDataReceived`.